internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -110,12 +122,7 @@ func expectedProxyContainers(infra *ir.ProxyInfra, deploymentConfig *egv1a1.Kube
 		}
 	}
 
-	var proxyMetrics *egv1a1.ProxyMetrics
-	if infra.Config != nil {
-		proxyMetrics = infra.Config.Spec.Telemetry.Metrics
-	}
-
-	if proxyMetrics != nil && proxyMetrics.Prometheus != nil {
+	if enablePrometheus(infra) {
```

Comment:
shouldnt this port always exist, since its the readiness port as well ?

Reply:
then we need to rename container name.
Keep it the same as before, will do it later.

Reply:
and think about https://github.com/envoyproxy/gateway/pull/1787